### PR TITLE
Limit max clients per workspace rule

### PR DESCRIPTION
**Pull request by LamprosPitsillos**
Superset of 
#3164

Almost all work done by @MightyPlaza 

TODO: 
Have a way to not switch to workspace after new window was moved from current.

### CODE REVIEW

**Comment by LamprosPitsillos** [comment]
```cpp
        else if ((delim = rule.find("maxclients:")) != std::string::npos)
            wsRule.maxClients = configStringToInt(rule.substr(delim + 11));
// Should i parse after this to check for `silent` ?
```
What would be better ? To change the type of `maxClients` to also store a bool if silent or make another variable ?

**Comment by MightyPlaza** [comment]
probably just make another var to store the bool

**Comment by LamprosPitsillos** [comment]
On issue i have is that new window takes focus, how can i disable that?

**Comment by MightyPlaza** [comment]
even with silent?

**Comment by LamprosPitsillos** [comment]
The workspace doesnt chance but the window does grab focus

I am now taking a look at
```cpp
void CKeybindManager::moveActiveToWorkspaceSilent(std::string args) {
```
I think the way to do it would be similar to this

**Comment by MightyPlaza** [comment]
oh, just pass the silent workspace rule to the already silent variable from the window rule

**Comment by LamprosPitsillos** [comment]
```cpp
 maxClients       = workspaceRule.maxClients;
            if (maxClients != 0 && maxClients < g_pCompositor->getVisibleWindowsOnWorkspace(pWorkspace->m_iID)) {
```
Is there a reson for these 2 lines running twice ? 
```
if (contition){
if (same contition){
}
}

```

**Comment by MightyPlaza** [comment]
pretty sure that can't be simplified

**Comment by LamprosPitsillos** [comment]
Is that how you solved the grouped/swallowed stuff?

**Comment by MightyPlaza** [comment]
wdym?

**Comment by LamprosPitsillos** [comment]
Is there anything else needed for this PR ?

File: src/events/Windows.cpp [hunk 1]
```diff
@@ -570,6 +572,43 @@ void Events::listener_mapWindow(void* owner, void* data) {
         }
     }
 
+    if (!isWorkspaceSet) {
+        auto       pWorkspace    = g_pCompositor->getWorkspaceByID(PWINDOW->m_iWorkspaceID);
+        auto       workspaceRule = pWorkspace ? g_pConfigManager->getWorkspaceRuleFor(pWorkspace) : SWorkspaceRule{};
+        const auto PMONITOR      = g_pCompositor->getMonitorFromID(PWINDOW->m_iMonitorID);
+
+        int        maxClients = workspaceRule.maxClients;
+        bool       silent     = workspaceRule.maxClientsSilent;
```
Comment (MightyPlaza) [inline src/events/Windows.cpp:581]: probably better to use a more descriptive name like `maxClientsSilent`

**Review by MightyPlaza — COMMENTED** [review]
also clang-format
rest lgtm

**Comment by LamprosPitsillos** [comment]
> also clang-format rest lgtm

is there an automated way to do this ? `make format` or something?

**Comment by MightyPlaza** [comment]
clang-format -i <file1> <file2>

**Comment by MightyPlaza** [comment]
@vaxerski

**Comment by vaxerski** [comment]
what? there's still a conflict. Request my review when ready2merge

**Comment by MightyPlaza** [comment]
pretty sure it was ready before
didn't notice it needed a rebase now

**Comment by LamprosPitsillos** [comment]
@MightyPlaza since you where the one who fixed the swallowing problem , could you give some insights on how to solve the conflicts please ?

**Comment by MightyPlaza** [comment]
`git pull --rebase`

**Comment by LamprosPitsillos** [comment]
> `git pull --rebase`

I was referring to what needs to stay and what need to leave , regarding the rebase.
Anw, i will give it a try on my own and test it after

**Comment by MightyPlaza** [comment]
I don't know what you did but it doesn't seem right
also conflicts still aren't resolved

**Comment by LamprosPitsillos** [comment]
> I don't know what you did but it doesn't seem right also conflicts still aren't resolved

Yeah... i have no idea what went wrong...

I wanted to fix them so i thought `git pull --rebase` and then solve the issue.
I synced the main branch of my fork with the "upstream" repo , and then i rebased my main and my branch.
After several rebases , i managed to rebase the main branch with my fork.
Checking back on my pull request, i see almost 100+ commits and the conflict still there...
Also checking back on my own fork , I noticed my branch was
This branch is 86 commits ahead, 76 commits behind hyprwm:main.

Am sorry,i have no idea what happened...
I will try and fix it

**Comment by MightyPlaza** [comment]
you're supposed to rebase your fork with the main branch
since it isn't a very big diff you could even just reset the branch and do the changes again then commit them and `git push --force`

**Pull request closed without merging**